packages/eslint-plugin/configs/esnext.js

```diff
@@ -0,0 +1,36 @@
+module.exports = {
+	env: {
+		es6: true,
+	},
+	extends: [
+		require.resolve( './es5.js' ),
+	],
+	parserOptions: {
+		sourceType: 'module',
+	},
+	rules: {
+		// Disable ES5-specific (extended from ES5)
+		'vars-on-top': 'off',
+
+		// Enable ESNext-specific
+		'array-bracket-spacing': [ 'error', 'always' ],
+		'arrow-parens': [ 'error', 'always' ],
+		'arrow-spacing': 'error',
+		'computed-property-spacing': [ 'error', 'always' ],
+		'constructor-super': 'error',
+		'no-const-assign': 'error',
+		'no-dupe-class-members': 'error',
+		'no-duplicate-imports': 'error',
+		'no-useless-computed-key': 'error',
+		'no-useless-constructor': 'error',
```

Comment:
Is `'no-var': 'error',` automatically applied to `es6` env? It's missing when you compare to the previous setup.